spec/controllers/api/v0x0/application_controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+RSpec.describe ApplicationController, :type => :request do
+  include SpecHelper
+
+  let(:tenant)          { Tenant.create!(:external_tenant => external_tenant) }
+  let(:source_type)     { SourceType.create!(:name => "openshift", :product_name => "OpenShift", :vendor => "Red Hat") }
+  let(:source)          { Source.create!(:source_type_id => source_type.id.to_s, :tenant_id => tenant.id , :name => "abc", :uid => "123") }
+  let(:external_tenant) { rand(1000).to_s }
+  let(:identity)        { Base64.encode64({'identity' => { 'account_number' => external_tenant}}.to_json) }
```

Comment:
@bdunne @syncrou When we get to the user stuff...this feels like a method that should live in ManageIQ::API::Common so we don't have to reinvent it everywhere.

Reply:
@Fryguy - Yes, I have a method called `ManageIQ::API::Common::Users.encode that does exactly this. Hoping to push up that PR later today.